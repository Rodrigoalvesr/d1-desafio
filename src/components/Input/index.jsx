import React from 'react';
import PropType from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import InputBase from '@material-ui/core/InputBase';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    outline: 'none',
    border: '1px solid #CCCFDE',
    padding: 6,
    borderRadius: 5,
  },
  icon: {
    display: 'flex',
    alignItems: 'center',
    marginRight: theme.spacing(1),
    marginLeft: theme.spacing(1),
    '& svg': {
      fill: theme.palette.primary.main,
    },
  },
  input: {
    fontSize: 12,
    '&::placeholder': {
      color: '#9196AB',
    },
  },
}));

function Input({ startIcon, className, ...rest }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {startIcon && <div className={classes.icon}>{startIcon}</div>}
      <InputBase
        fullWidth
        className={classnames(classes.input, className)}
        {...rest}
      />
    </div>
  );
}

Input.propTypes = {
  startIcon: PropType.node,
  className: PropType.string,
};

export default Input;
