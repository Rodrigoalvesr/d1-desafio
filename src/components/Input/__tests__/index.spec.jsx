import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../index';

describe('Input', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Input />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('Renders correctly with icon', () => {
    const component = renderer.create(<Input startIcon={<div />} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
