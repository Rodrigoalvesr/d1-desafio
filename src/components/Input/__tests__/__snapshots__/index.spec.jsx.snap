// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiInputBase-root makeStyles-input-3 MuiInputBase-fullWidth"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;

exports[`Input Renders correctly with icon 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="makeStyles-icon-2"
  >
    <div />
  </div>
  <div
    className="MuiInputBase-root makeStyles-input-3 MuiInputBase-fullWidth"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
    />
  </div>
</div>
`;
